core/src/main/scala/sttp/tapir/server/interpreter/EncodeOutputs.scala

```diff
@@ -59,19 +61,41 @@ class EncodeOutputs[B, S](rawToResponseBody: ToResponseBody[B, S]) {
         )
       case EndpointOutput.OneOf(mappings, _) =>
         val enc = encoded[Any]
-        val mapping = mappings
-          .find(mapping => mapping.appliesTo(enc))
-          .getOrElse(throw new IllegalArgumentException(s"No status code mapping for value: $enc, in output: $output"))
-        apply(mapping.output, ParamsAsAny(enc), mapping.statusCode.map(ov.withStatusCode).getOrElse(ov))
+
+        val bodyMappings: Map[MediaType, StatusMapping[_]] = mappings
+          .filter(_.appliesTo(enc))
+          .flatMap(sm =>
+            sm.output.traverseOutputs {
+              case EndpointIO.Body(bodyType, codec, _) =>
+                Vector[(MediaType, StatusMapping[_])](
+                  charset(bodyType).map(ch => codec.format.mediaType.charset(ch.name())).getOrElse(codec.format.mediaType) -> sm
+                )
+              case EndpointIO.StreamBodyWrapper(StreamBodyIO(_, codec, _, charset)) =>
+                Vector[(MediaType, StatusMapping[_])](
+                  charset.map(ch => codec.format.mediaType.charset(ch.name())).getOrElse(codec.format.mediaType) -> sm
+                )
+              case EndpointIO.Empty(codec, _) => Vector[(MediaType, StatusMapping[_])](codec.format.mediaType -> sm)
```

Comment:
I'm not sure this will work as expected. The media type for an empty output will always be `text/plain` (by definition). An empty output should match the no-media-type, so we'd get a `Map[Option[MediaType], StatusMapping[_]]`

Reply:
And we can't really depend on `Empty` here. The output might contain headers, but not be empty.